exec/exec.go

```diff
@@ -42,14 +43,23 @@ func newCmdError(args string, cause error, stderr string) *CmdError {
 	return &CmdError{Args: args, Stderr: stderr, Cause: cause}
 }
 
+// Behavior when command takes longer than the passed in timeout to finish
+// By default, SIGKILL is sent and the process isn't waited upon
+type TimeoutBehavior struct {
+	Signal     syscall.Signal // signal to send to the process
+	ShouldWait bool           // whether to wait for the command to exit once timeout is reached
+}
+
 type CmdOpts struct {
-	Timeout  time.Duration
-	Redactor func(text string) string
+	Timeout         time.Duration            // how long to wait for the command to exit
+	Redactor        func(text string) string // function to redact tokens from the output
+	TimeoutBehavior TimeoutBehavior          // what to do in case of timeout
```

Comment:
```suggestion
	// Timeout determines how long to wait for the command to exit
	Timeout         time.Duration
	// Recactor redacts tokens from the output
	Redactor        func(text string) string
	// TimeoutBehavior configures what to do in case of timeout
	TimeoutBehavior TimeoutBehavior
```